src/EditorFeatures/CSharpTest/InlineTypeCheck/CSharpInlineTypeCheckTests.cs

```diff
@@ -0,0 +1,367 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.InlineTypeCheck;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.UseImplicitType;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.InlineTypeCheck
+{
+    public partial class CSharpInlineTypeCheckTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return new Tuple<DiagnosticAnalyzer, CodeFixProvider>(
+                new CSharpInlineTypeCheckDiagnosticAnalyzer(),
+                new CSharpInlineTypeCheckCodeFixProvider());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheck1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|var|] x = o as string;
+        if (x != null)
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (o is string x)
+        {
+        } 
+    }
+}");
+        }
```

Comment:
> Should these also test code that's semantically correct? After all, o is not defined.

I don't think so no.  The code, syntactically, completely conveys the intent of the user.  Semantics should only be necessary if we'd make a different sort of decision based on what semantics told us.